solr/core/src/java/org/apache/solr/handler/admin/api/SnapshotBackupAPI.java

```diff
@@ -0,0 +1,174 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.handler.admin.api;
+
+import static org.apache.solr.client.solrj.impl.BinaryResponseParser.BINARY_CONTENT_TYPE_V2;
+import static org.apache.solr.handler.ReplicationHandler.ERR_STATUS;
+import static org.apache.solr.security.PermissionNameProvider.Name.CORE_EDIT_PERM;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import java.lang.invoke.MethodHandles;
+import java.util.function.Consumer;
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import org.apache.solr.api.JerseyResource;
+import org.apache.solr.client.api.model.SolrJerseyResponse;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.annotation.JsonProperty;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.core.SolrCore;
+import org.apache.solr.handler.ReplicationHandler;
+import org.apache.solr.handler.ReplicationHandler.ReplicationHandlerConfig;
+import org.apache.solr.jersey.JacksonReflectMapWriter;
+import org.apache.solr.jersey.PermissionName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** V2 endpoint for Backup API used for User-Managed clusters and Single-Node Installation. */
+@Path("/cores/{coreName}/replication/backups")
+public class SnapshotBackupAPI extends JerseyResource {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final SolrCore solrCore;
+  private final ReplicationHandlerConfig replicationHandlerConfig;
+
+  @Inject
+  public SnapshotBackupAPI(SolrCore solrCore, ReplicationHandlerConfig replicationHandlerConfig) {
+    this.solrCore = solrCore;
+    this.replicationHandlerConfig = replicationHandlerConfig;
+  }
+
+  /**
+   * This API (POST /api/cores/coreName/replication/backups {...}) is analogous to the v1
+   * /solr/coreName/replication?command=backup
+   */
+  @POST
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
+  @Operation(summary = "Backup command using ReplicationHandler")
+  @PermissionName(CORE_EDIT_PERM)
+  public BackupReplicationResponse createBackup(
+      @RequestBody BackupReplicationRequestBody backupReplicationPayload) throws Exception {
+    if (backupReplicationPayload == null) {
```

Comment:
[0] It comes up often enough that maybe it's worth moving this check-null-and-throw to be a protected method on `JerseyResource` similar to the existing `ensureRequiredParameterProvided`.

Reply:
makes sense, method added.